cabot/cabotapp/models.py

```diff
@@ -489,28 +489,35 @@ def should_run(self):
         next_run_time = self.last_run + timedelta(minutes=self.frequency)
         return timezone.now() > next_run_time
 
+    @transaction.atomic()
     def run(self):
         start = timezone.now()
         try:
-            result = self._run()
-        except SoftTimeLimitExceeded as e:
-            result = StatusCheckResult(check=self)
-            result.error = u'Error in performing check: Celery soft time limit exceeded'
-            result.succeeded = False
+            result, tags = self._run()
+        except SoftTimeLimitExceeded:
+            result = StatusCheckResult(check=self, succeeded=False,
+                                       error=u'Error in performing check: Celery soft time limit exceeded')
+            tags = ['celery_timeout']
         except Exception as e:
-            result = StatusCheckResult(check=self)
-            result.error = u'Error in performing check: %s' % (e,)
-            result.succeeded = False
+            result = StatusCheckResult(check=self, succeeded=False, error=u'Error in performing check: %s' % (e,))
+            tags = ['run_error']
+
         finish = timezone.now()
         result.time = start
         result.time_complete = finish
         result.save()
+
+        # would like to do this in bulk, but django can't fetch the tag object PKs from bulk_create()...
```

Comment:
What's the max # of tags you'd anticipate? if there's a possibility of a really high number of tags this might be something we'd want to think about more, otherwise seems ok

Reply:
Probably 1-3 tags. The real problem is that `bulk_create()` doesn't have a way to do `INSERT IGNORE`...[until Django 2.2](https://stackoverflow.com/a/54189488). The upgrades never end.

Reply:
ok one by one seems fine then :)